python/ray/tune/tests/test_result_grid.py

```diff
@@ -100,6 +154,32 @@ def f(config):
     assert best_result.metrics["x"] == 2
 
 
+def test_best_result_checkpoint_history(ray_start_2_cpus):
+    def f(config):
+        for i in range(2):
+            with tune.checkpoint_dir(step=i) as checkpoint_dir:
+                path = os.path.join(checkpoint_dir, "checkpoint")
+                with open(path, "w") as f:
+                    f.write(json.dumps(dict(x=config["x"], step=i)))
+            tune.report(x=config["x"], step=i)
+
+    analysis = tune.run(f, config={"x": tune.grid_search([1, 3])})
```

Comment:
out of scope here but we should really find a way to construct experiment checkpoints without always running full runs for testing.